tools/diagnose.py

```diff
@@ -171,6 +161,31 @@ def check_network(args):
     for name, url in URLS.items():
         test_connection(name, url, args.timeout)
 
+
+def check_environment():
+    print('----------Environment----------')
+    for k,v in os.environ.items():
+        print('{}="{}"'.format(k,v))
+
+
+def parse_args():
```

Comment:
why is this moved below?

Reply:
usually you want it next to main, instead of scrolling all the way up.